torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py

```diff
@@ -275,6 +279,11 @@ def aot_dispatch_autograd_graph(
     torch._dynamo.utils.assert_no_fake_params_or_buffers(fx_g)
     fx_g.graph.eliminate_dead_code()
     fx_g.recompile()
+
+    # Copy metadata from fw to bw nodes
+    copy_fwd_metadata_to_bw_nodes(fx_g)
+    fx_g.recompile()
```

Comment:
There's no point in recompiling, mutating and recompiling again. Just recompile it once at the end.